judge/bridge/judgelist.py

```diff
@@ -98,14 +98,18 @@ def abort(self, submission):
     def check_priority(self, priority):
         return 0 <= priority < self.priorities
 
-    def judge(self, id, problem, language, source, priority):
+    def judge(self, id, problem, language, source, judge_id, priority):
         with self.lock:
             if id in self.submission_map or id in self.node_map:
                 # Already judging, don't queue again. This can happen during batch rejudges, rejudges should be
                 # idempotent.
                 return
 
-            candidates = [judge for judge in self.judges if not judge.working and judge.can_judge(problem, language)]
+            candidates = [
+                judge for judge in self.judges
+                if not judge.working and judge.can_judge(problem, language) and
+                    (judge_id is None or judge.judge.name == judge_id)
+            ]
             logger.info('Free judges: %d', len(candidates))
```

Comment:
Should probably log a slightly different message in case of `judge_id` being set